docs/design-docs/07-package-orchestration.md

```diff
@@ -0,0 +1,491 @@
+
+## CaD Concepts Elaborated
+
+Concepts briefly introduced above are elaborated in more detail in this section.
+
+### Repositories
+
+[Config Sync][] and [kpt][] currently integrate with [git][] repositories, and
+there is an existing design to add [OCI support](./02-oci-support.md) to kpt.
+Initially, the CaD Core system will prioritize integration with [git][], and
+support for addition repository types may be added in the future as required.
+
+Requirements applicable to all repositories include: ability to store packages,
+their versions, and sufficient metadata associated with package to capture:
+
+* package dependency relationships (upstream - downstream)
+* package lifecycle state (draft, proposed, published)
+* package purpose (base package)
+* (optionally) even customer-defined attributes
+
+At repository registration, customers must be able to specify details needed to
+store packages in appropriate locations in the repository. For example,
+registration of a Git repository must accept branch and a directory.
+
+Repositories may have associated guardrails - mutation and validation functions
+that ensure and enforce requirements of all packages in the repository,
+including gating promotion of a package to a *published* lifecycle stage.
+
+_Note_: A user role with sufficient permissions can register a package or
+function repository, including repositories containing functions authored by
+the cutomer, or other providers. Since the functions in the registered
+repositories become discoverable, customers must be aware of the implications of
+registering function repositories and trust the contents thereof.
+
+### Package Versioning
+
+Packages are sequentially versioned. The important requirements are:
+
+* ability to compare any 2 versions of a packge to be either "newer than",
+  equal, or "older than" relationship
+* ability to support automatic assignment of versions
+* ability to support [optimistic concurrency][optimistic-concurrency] of package
+  changes via version numbers
+* simple model which easily supports automation
+
+We plan to use a simple integer sequence to represent package versions.
+
+### Package Relationships
+
+Kpt packages support the concept of ***upstream***. When a package is cloned
+from another, the new package (called ***downstream*** package) maintains an
+upstream link to the specific version of the package from which it was cloned.
+If a new version of the upstream package becomes available, the upstream link
+can be used to [update](https://kpt.dev/book/03-packages/05-updating-a-package)
+the downstream package.
+
+### Deployment
+
+[Config Sync][] is the deployment mechanism used by CaD Core implementation.
+Here we highlight some key attributes of the deployment mechanism and its
+integration within the CaD Core:
+
+* _Published_ packages in a deployment repository are considered ready to be
+  deployed
+* It should be possible (and straightforward) to deploy individual packages and
```

Comment:
Is this work that is completed or something we want to support? The wording here suggest the latter while the document up until here seems to describe what is currently available.

Reply:
updated. hopefully this is better. LMK